src/amount.cpp

```diff
@@ -30,3 +31,16 @@ std::string CFeeRate::ToString() const
 {
     return strprintf("%d.%08d %s/kB", nSatoshisPerK / COIN, nSatoshisPerK % COIN, CURRENCY_UNIT);
 }
+
+
+CAmount CFeeRate::GetDust() const
+{
+    CAmount dust = txDust.Value();
+    // If dust has not been configured, then
+    // "Dust" is defined in terms of CTransaction::minRelayTxFee, which has units satoshis-per-kilobyte.
+    // If you'd pay more than 1/3 in fees to spend something, then we consider it dust.
+    // A typical spendable txout is 34 bytes big, and will need a CTxIn of at least 148 bytes to spend:
+    if (dust == 0)
+        dust = 3 * minRelayTxFee.GetFee(TYPICAL_UTXO_LIFECYCLE_SIZE);
+    return dust;
+}
```

Comment:
we already have something in place for that, see `GetDustThreshold()` in `src/primitives/transaction.h` line 151.

We also have a tweak for set dust threshold `net.dustThreshold`
```c++
CTweak<unsigned int> nDustThreshold("net.dustThreshold", "Dust Threshold (in satoshis).", DEFAULT_DUST_THRESHOLD);
```

Reply:
great observation, thanks!

Reply:
ok,that tweak is the relay policy on the minimum value an output may have.  What I am using here is a different but related concept -- its basically saying when should the wallet put surplus into the fee rather than creating a new output.  This could be a lot larger than the relay dust limit but it cannot be any smaller than the relay policy "dust".  I will change the code to capture this logic.